docs/introduction-to-smart-contracts.rst

```diff
@@ -504,10 +511,8 @@ Delegatecall / Callcode and Libraries
 
 There exists a special variant of a message call, named **delegatecall**
 which is identical to a message call apart from the fact that
-the code at the target address is executed in the context of the calling
-contract. In practice this means ``msg.sender`` and ``msg.value`` retain
-the values that were passed to the delegator, while executing code that
-lives in the delegated contract.
+the code at the target address is executed in the context (i.e. at the address) of the calling
+contract and ``msg.sender`` and ``msg.value`` do not change thair values.
 
 This means that a contract can dynamically load code from a different
 address at runtime. Storage, current address and balance still
```

Comment:
```suggestion
This means that a contract can dynamically load code from a different
address at runtime and execute it as if it was one of its internal functions. Storage, current address and balance still
```

Maybe that's a good way to clarify what you wanted to have in https://github.com/ethereum/solidity/pull/12910/files#r845484979 ?

Reply:
Sounds good to me. I'll add that to the PR then.

Reply:
On the other hand "as if it was one of its internal functions" might read like the called contract would somehow magically get access to the symbols in the calling contracts scope... but maybe I'm just slowly going mad over thinking about docs ;-).

Reply:
Maybe. I thought it still gives you a better picture of what will happen.

But yeah, could be misleading. Let's just merge it as is with only typos fixes.

Reply:
Yeah, I'm not sure anymore, maybe I'm starting to overthink what stuff might imply or not ;-). I'm fine with adding this and with not adding this.